### PR TITLE
Make layer cue-able

### PR DESCRIPTION
**Pull request by rettinghaus**
Usually in parts a whole layer gives the cue, so it seems to make sense to make `layer` part of `att.cue`. 
image
(Example taken from _Behind Bars_, p 572)

### CODE REVIEW

**Review by craigsapp — APPROVED** [review]
This is a useful feature to have.  The example music demonstrates that most cues in orchestral parts are encoded in layers, with a regular-sized full-measure rests in another layer to reinforce that the cue notes are not to be played by the performer.

**Review by kepper — CHANGES_REQUESTED** [review]
This sounds perfectly reasonable, and I'm not requesting a change to the one line added here. But. There should be some documentation about this, like we just discussed at the ODD Friday. It needs to be checked how much actual documentation we have in the Guidelines about cue'd things…

**Comment by musicEnfanthen** [comment]
See also #668

**Comment by bwbohl** [comment]
2022-01-28 ODD meeting: @rettinghaus if you could please be so kind as to add some documentation to the guidelines. Apparently there isn't anything about `@cue` so far, some general example might be a first start, even when limiting it to layer-cueing.

**Comment by rettinghaus** [comment]
Added a section about cue notes in the Guidelines.

File: source/examples/cmn/cmn-cue_layer.txt
Comment (ahankinson) [inline source/examples/cmn/cmn-cue_layer.txt:28]: I'm wondering if we should recommend that `cue` should also use `sameas` and point to the cued layer, rather than encoding the layer twice. Thoughts?

Comment (rettinghaus) [reply source/examples/cmn/cmn-cue_layer.txt:28]: Sorry, I don't understand.

Comment (kepper) [reply source/examples/cmn/cmn-cue_layer.txt:28]: I've run into this recently. I would say it should be optional, but not necessarily recommended. The cue notes / layer is still an entity in itself, even though it has a semantic connection to some other notes / layer and is derived from there. It could have incorrect notes, though. In that case, one may perhaps prefer a `@corresp` over `@sameas`?

Comment (ahankinson) [reply source/examples/cmn/cmn-cue_layer.txt:28]: The notes in a cue layer correspond to notes in another layer, correct? They're not played, but they're there to indicate that you should listen to those notes for your own entrance.

So should we use some of the available mechanisms (sameas, corresp, etc.) to indicate that this link exists, rather than to encode them as completely separate notes?

Comment (ahankinson) [reply source/examples/cmn/cmn-cue_layer.txt:28]: In the case of a cued layer, I would expect that a full layer would be cued. Otherwise, we would just mark four notes as cues.

Comment (rettinghaus) [reply source/examples/cmn/cmn-cue_layer.txt:28]: Yes, we can point to the original layer, if we have a full score encoded. If we have only a single part, where should it point at?

Comment (ahankinson) [reply source/examples/cmn/cmn-cue_layer.txt:28]: Your addition looks great! Thanks.

**Review by ahankinson — APPROVED** [review]

**Comment by ahankinson** [comment]
Probably should have one more reviewer before merging. Since @craigsapp reviewed it there have been some changes.

**Comment by musicEnfanthen** [comment]
Would it be helpful to include an example of the proposed linking mechanism for cued layers with `corresp` in chapter [10.1.1](https://music-encoding.org/guidelines/dev/content/analysisharm.html#analysisDescribingRelationships) of the guidelines and link there from the description here?

File: source/docs/04-cmn.xml
Comment (bwbohl) [inline source/docs/04-cmn.xml]: instruments/singers --> instruments or singers?

**Review by bwbohl — COMMENTED** [review]
only a minor comment

many thanks @rettinghaus

**Review by bwbohl — APPROVED** [review]

**Comment by bwbohl** [comment]
@kepper could you revise your review please

File: source/docs/04-cmn.xml
Comment (musicEnfanthen) [inline source/docs/04-cmn.xml:1131]: Would it be possible to include another encoding example here for this linking mechanism with `corresp`. This would be very helpful for users consulting this passage.

Comment (rettinghaus) [reply source/docs/04-cmn.xml:1131]: As you suggested this should probably go into [chapter 13](https://music-encoding.org/guidelines/dev/content/analysisharm.html#analysisDescribingRelationships) where nothing can be found about `@corresp`at all. Opened #915.

Comment (musicEnfanthen) [reply source/docs/04-cmn.xml:1131]: `@corresp` itself and all other attributes of `att.linking` are momentarily explained in chapter 10 (Analytical markup). If that's a good place for them, we should discuss over there in #915, right. 

For the passage here, I am not asking to explain `corresp` in general (what would be too much out of scope of this PR), just to add a minimal encoding example how you would target the "sounding counterpart" from the cued layer. (My initial thought was to link to the corresp section from here, but since there is nothing really helpful for the use case here, it seems easier for now to include a small example here which exemplifies that usage of cued layers.) What do you think?

**Comment by craigsapp** [comment]
Here is another example that could use this feature that I have come across:

Screen Shot 2022-02-21 at 1 34 30 PM

From Stravinsky: Perséphone (Melodrama in Three Scenes for Solo Tenor, Choir and Orchestra)

The second staff (violin 1) has two layers, with the first one being cue-sized showing the sounding pitches (also in square brackets).   The performance fingerings are in the second layer.

**Review by kepper — APPROVED** [review]
:)

**Comment by kepper** [comment]
by instruction of our technical co-chairs during ODD meeting today – many thanks ;-)

**Pull request merged**
